flink-yarn/src/main/java/org/apache/flink/yarn/YarnApplicationMasterRunner.java

```diff
@@ -401,11 +406,14 @@ protected int runApplicationMaster(Configuration config) {
 			}
 
 			if (actorSystem != null) {
-				try {
-					actorSystem.shutdown();
-				} catch (Throwable tt) {
-					LOG.error("Error shutting down actor system", tt);
-				}
+				actorSystem.terminate().onComplete(
+					new OnComplete<Terminated>() {
+						public void onComplete(Throwable failure, Terminated result) {
+							if (failure != null) {
+								LOG.error("Error shutting down actor system", failure);
+							}
+						}
+					}, org.apache.flink.runtime.concurrent.Executors.directExecutionContext());
```

Comment:
This PR looks good to me. Here are a little questions, could this be simplified by `import static`?